src/services/symbolication.rs

```diff
@@ -1605,6 +1608,116 @@ impl SymbolicationActor {
         future::join_all(futures).await
     }
 
+    /// Post-processes breakpad's stackwalking result to extract module list and stacktraces.
+    ///
+    /// The breakpad [`ProcessState`] contains the results of the breakpad stackwalking.  This
+    /// code extracts from this the module list as well as the stack traces in the format required
+    /// for the symbolication response on the API.
+    fn post_process(
+        process_state: ProcessState,
+        cfi_caches: CfiCacheModules,
+    ) -> StackwalkingOutput {
+        let minidump_state = MinidumpState::new(&process_state);
+
+        // Start building the module list for the symbolication response.
+        let mut module_builder =
+            ModuleListBuilder::new(cfi_caches, &process_state, minidump_state.object_type());
+
+        // Finally iterate through the threads and build the stacktraces to
+        // return, marking modules as used when they are referenced by a frame.
+        let requesting_thread_index: Option<usize> =
+            process_state.requesting_thread().try_into().ok();
+        let threads = process_state.threads();
+        let mut stacktraces = Vec::with_capacity(threads.len());
+        for (index, thread) in threads.iter().enumerate() {
+            let registers = match thread.frames().get(0) {
+                Some(frame) => {
+                    map_symbolic_registers(frame.registers(minidump_state.system_info.cpu_arch))
+                }
+                None => Registers::new(),
+            };
+
+            // Trim infinite recursions explicitly because those do not
+            // correlate to minidump size. Every other kind of bloated
+            // input data we know is already trimmed/rejected by raw
+            // byte size alone.
+            let frame_count = thread.frames().len().min(20000);
+            let mut frames = Vec::with_capacity(frame_count);
+            for frame in thread.frames().iter().take(frame_count) {
+                let return_address = frame.return_address(minidump_state.system_info.cpu_arch);
+                module_builder.mark_referenced(return_address);
+
+                frames.push(RawFrame {
+                    instruction_addr: HexValue(return_address),
+                    package: frame.module().map(CodeModule::code_file),
+                    trust: frame.trust(),
+                    ..RawFrame::default()
+                });
+            }
+            stacktraces.push(RawStacktrace {
+                is_requesting: requesting_thread_index.map(|r| r == index),
+                thread_id: Some(thread.thread_id().into()),
+                registers,
+                frames,
+            });
+        }
+
+        (module_builder.build(), stacktraces, minidump_state)
+    }
+
+    fn stackwalk_name_subject_to_change(
```

Comment:
Thoughts on the name of this method (and if it should even be a method in the first place, as opposed to a closure in `stackwalk_minidump_with_cfi`)?

Reply:
`procspawn_inner_stackwalk`?  Though I'm also fine I think with leaving it as a closure I think.

Reply:
Yeah, I'm getting the feeling that aside from the formatting problem this doesn't need to be a function.

Reply:
ooh, rustfmt works with this as a function?  I guess that's enough reason to keep it as a function.